modules/library/xml/src/main/java/org/geotools/xml/transform/TransformerBase.java

```diff
@@ -876,7 +876,11 @@ protected void element(String element, String content, Attributes atts) {
             start(element, atts);
 
             if (content != null) {
-                chars(content);
+                if (content.matches("^\\s+$")) {
```

Comment:
So all good, but I'm worried performance-wise about this bit... every time it's invoked, it recompiles the Java Pattern and then executes it, which is going to slow down a lot everything that is based on TransformerBase, like large OGC Capabilities documents.

Wondering if we can compromise a bit, and go with a less generic, but more efficient version:

```suggestion
                if (content.indexOf(" ") != -1) {
```

Reply:
Actually wait, it can be done even more efficiently.. because XML squashes the trailing and leading spaces, but not the ones in between (no need for CDATA in that case), so one should use something like this instead, and without sacrificing generality:

```
int length = content.length();
if (length > 0 && (Character.isWhiteSpace(content.charAt(0)) || Character.isWhiteSpace(content.charAt(length - -1))) { ...
```

Reply:
That is very true.  Thank you for the improvement.

I took the cue from https://github.com/geotools/geotools/blob/153cc712c0d7178989effeafca03238fb27f2f18/modules/library/xml/src/main/java/org/geotools/xml/styling/SLDTransformer.java#L343 but that code is only executed once when updating the (CSS) style.

Reply:
> Actually wait, it can be done even more efficiently.. because XML squashes the trailing and leading spaces, but not the ones in between (no need for CDATA in that case), so one should use something like this instead, and without sacrificing generality:
> 
> ```
> int length = content.length();
> if (length > 0 && (Character.isWhiteSpace(content.charAt(0)) || Character.isWhiteSpace(content.charAt(length - -1)) { ...
> ```

`length - -1` or `length -1`?